test/jdk/javax/swing/JButton/SwingButtonResizeTestWithOpenGL.java

```diff
@@ -0,0 +1,311 @@
+ * with swing components after window is zoomed to maximum size and then
+ * resized back to normal.  The test case simulates this operation using
+ * a JButton.  A file image of the component will be saved before and after
+ * window resize. The test passes if both button images are the same.
+ * @run main/othervm -Dsun.java2d.opengl=True -Dsun.java2d.opengl.fbobject=false SwingButtonResizeTestWithOpenGL
+ * @run main/othervm SwingButtonResizeTestWithOpenGL
+ */
+
+public class SwingButtonResizeTestWithOpenGL {
+    private static Robot robot;
+    private static CountDownLatch focusGainedLatch;
+    private JFrame frame;
+    private JButton button;
+
+    public SwingButtonResizeTestWithOpenGL() {
+
+        try {
+            SwingUtilities.invokeAndWait(() -> createGUI());
+        } catch (Exception e) {
+            throw new RuntimeException("Problems creating GUI");
+        }
+    }
+
+    private void createGUI() {
+        frame = new JFrame("SwingButtonResizeTestWithOpenGL");
+        button = new JButton("Button A");
+        frame.setLocationRelativeTo(null);
+        frame.setLocation(200, 200);
+        frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        button.setPreferredSize(new Dimension(300, 300));
+        button.addFocusListener(new FocusAdapter() {
+            public void focusGained(FocusEvent fe) {
+                focusGainedLatch.countDown();
+            }
+        });
+        frame.getContentPane().setLayout(new FlowLayout());
+        frame.getContentPane().add(button);
+        frame.pack();
+        frame.setVisible(true);
+        frame.toFront();
+    }
+
+    public static void main(String[] args) throws Exception {
+        focusGainedLatch = new CountDownLatch(1);
+        SwingButtonResizeTestWithOpenGL test = new SwingButtonResizeTestWithOpenGL();
+        test.runTest();
+    }
+
+    public void runTest() throws Exception {
+        BufferedImage bimage1;
+        BufferedImage bimage2;
+
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+            robot.setAutoDelay(200);
+
+            if (focusGainedLatch.await(3, TimeUnit.SECONDS)) {
+                System.out.println("Button focus gained...");
+            } else {
+                System.out.println("Button focus not gained...");
+                throw new RuntimeException("Can't gain focus on button even after waiting too long..");
+            }
+
+            System.out.println("Getting initial button image..image1");
+            bimage1 = getButtonImage();
+
+            File file1 = new File("image1.png");
+            saveButtonImage(bimage1, file1);
+
+            //some platforms may not support maximize frame
+            if (frame.getToolkit().isFrameStateSupported(JFrame.MAXIMIZED_BOTH)) {
+
+                robot.waitForIdle();
+
+                frame.setExtendedState(JFrame.MAXIMIZED_BOTH);  //maximize
+                // frame from normal size
```

Comment:
```suggestion
            // some platforms may not support maximize frame
            if (frame.getToolkit().isFrameStateSupported(JFrame.MAXIMIZED_BOTH)) {

                robot.waitForIdle();

                frame.setExtendedState(JFrame.MAXIMIZED_BOTH);  //maximize
                // frame from normal size
```

I see some of your comments with and without a space after the `//`. Maybe standardize this to have the space.